libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -106,9 +112,16 @@ class MapboxNavigation(
     private val directionsSession: DirectionsSession
     private val tripService: TripService
     private val tripSession: TripSession
-    private val navigationSession = NavigationSession(context)
+    private val navigationSession = MapboxNavigationSession(context)
     private val internalRoutesObserver = createInternalRoutesObserver()
     private val internalOffRouteObserver = createInternalOffRouteObserver()
+    private val internalLocationObserver = createInternalLocationObserver()
+    private val internalRouteProgressObserver = createInternalRouteProgressObserver()
+    private val internalNavigationSessionObserver = createInternalNavigationSessionObserver()
+    private val fasterRouteTimer: MapboxTimer
+
+    private lateinit var enhancedLocation: Location
```

Comment:
We don't need this field and the observer that assigns it, we can always call `TripSession#tripSession.getEnhancedLocation()`.